core/src/main/java/org/elasticsearch/index/analysis/SynonymTokenFilterFactory.java

```diff
@@ -23,86 +23,165 @@
 import org.apache.lucene.analysis.LowerCaseFilter;
 import org.apache.lucene.analysis.TokenStream;
 import org.apache.lucene.analysis.Tokenizer;
-import org.apache.lucene.analysis.core.WhitespaceTokenizer;
 import org.apache.lucene.analysis.synonym.SolrSynonymParser;
 import org.apache.lucene.analysis.synonym.SynonymFilter;
 import org.apache.lucene.analysis.synonym.SynonymMap;
 import org.apache.lucene.analysis.synonym.WordnetSynonymParser;
+import org.elasticsearch.Version;
 import org.elasticsearch.common.io.FastStringReader;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.env.Environment;
 import org.elasticsearch.index.IndexSettings;
 import org.elasticsearch.indices.analysis.AnalysisModule;
 
+import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.Reader;
 import java.util.List;
 
 public class SynonymTokenFilterFactory extends AbstractTokenFilterFactory {
 
-    protected final SynonymMap synonymMap;
+    /**
+     * @deprecated this property only works with tokenizer property
+     */
+    @Deprecated
     protected final boolean ignoreCase;
+    protected final String format;
+    protected final boolean expand;
+    protected final Settings settings;
 
     public SynonymTokenFilterFactory(IndexSettings indexSettings, Environment env, AnalysisRegistry analysisRegistry,
                                       String name, Settings settings) throws IOException {
         super(indexSettings, name, settings);
+        this.settings = settings;
 
-        Reader rulesReader = null;
+        this.ignoreCase =
+            settings.getAsBooleanLenientForPreEs6Indices(indexSettings.getIndexVersionCreated(), "ignore_case", false, deprecationLogger);
+        if (settings.get("ignore_case") != null) {
+            deprecationLogger.deprecated(
+                "This tokenize synonyms with whatever tokenizer and token filters appear before it in the chain. " +
+                "If you need ignore case with this filter, you should set lowercase filter before this");
+        }
+
+        this.expand =
+            settings.getAsBooleanLenientForPreEs6Indices(indexSettings.getIndexVersionCreated(), "expand", true, deprecationLogger);
+
+        // for backward compatibility
+        if (indexSettings.getIndexVersionCreated().before(Version.V_6_0_0_alpha3)) {
+            deprecationLogger.deprecated(
+                "This filter tokenize synonyms with whatever tokenizer and token filters appear before it in the chain.");
```

Comment:
This is the wrong place? It should be in the `else` block if `settings.get("tokenizer")` is not null?